### PR TITLE
Fix various FreeBSD build-breakers.

### PR DESCRIPTION
**Pull request by josteink**
Fixes various issues with things not OSX being treated as if it's Linux.

While generally a decent strategy, it does not always apply and these specific issues are breaking the build on FreeBSD.

### CODE REVIEW

File: src/pal/src/loader/module.cpp [hunk 1]
```diff
@@ -1382,6 +1382,8 @@ static HMODULE LOADLoadLibrary(LPCSTR ShortAsciiName, BOOL fDynamic)
     {
 #if defined(__APPLE__)
         ShortAsciiName = "libc.dylib";
+#elif defined(__FreeBSD__)
+	ShortAsciiName = "libc.so";
```
Comment (kangaroo) [inline src/pal/src/loader/module.cpp]: Please fix the whitespace here.

File: src/pal/src/misc/sysinfo.cpp [hunk 2]
```diff
@@ -256,8 +256,12 @@ GlobalMemoryStatusEx(
     // We do this only when we have the total physical memory available.
     if (lpBuffer->ullTotalPhys > 0)
     {
-#if HAVE_SYSCONF && HAVE__SC_AVPHYS_PAGES
+#if HAVE_SYSCONF
+#if HAVE__SC_AVPHYS_PAGES
         lpBuffer->ullAvailPhys = sysconf(_SC_AVPHYS_PAGES) * sysconf(_SC_PAGE_SIZE);
+#else
+        lpBuffer->ullAvailPhys = sysconf(_SC_PHYS_PAGES) * sysconf(_SC_PAGE_SIZE);
```
Comment (kangaroo) [inline src/pal/src/misc/sysinfo.cpp]: This needs a HAVE__SC_PHYS_PAGES configure.cmake check, and guard.

File: src/pal/src/loader/module.cpp (hunk 1 above)
Comment (akoeplinger) [inline src/pal/src/loader/module.cpp]: should be using spaces for indentation

File: src/pal/src/misc/sysinfo.cpp (hunk 2 above)
Comment (josteink) [inline src/pal/src/misc/sysinfo.cpp]: Good point and good catch. Will fix that up.

**Comment by josteink** [comment]
Pushed a new commit which should address both concerns. `HAVE__SC_PHYS_PAGES` was already defined in `configure.cmake`.

**Comment by josteink** [comment]
@janvorli  I've implemented the final changes you suggested and it does indeed look better and more explicit.

Hopefully it should be good enough for merging at this point? :)

**Comment by janvorli** [comment]
LGTM, thank you very much for making all the changes.

**Comment by kangaroo** [comment]
LGTM

**Pull request merged**